src/webui/webapplication.cpp

```diff
@@ -265,8 +265,11 @@ void WebApplication::doProcessRequest()
     for (const Http::UploadedFile &torrent : request().files)
         data[torrent.filename] = torrent.data;
 
+    const ExecutionContext context = ((request().method == Http::METHOD_GET)
+                                    ? ExecutionContext::Safe : ExecutionContext::Unsafe);
```

Comment:
IMO, safe/unsafe should have different meaning for actions and execution contexts. 
1. "Safe" action is not "dangerous" so it can be accessed in any way (even in "unsafe" context). "Unsafe" action can be "dangerous" (e.g. it require some private/confident data) so it's required to be accessed in "safe" context only.
2. "Unsafe" context doesn't provide any protection (e.g. HTTP GET request) so only "safe" actions can be accessed within it. "Safe" context provide some additional protection so it allow both "safe" and "unsafe" actions.

Reply:
> IMO, safe/unsafe should have different meaning for actions and execution contexts.


You keep pushing for this but this is not how web servers are implemented. If an action requires a GET request, you must use a GET. You can't also use a POST just because it's safe to do so.

Reply:
I won't argue... but, IIRC, POST can be used for those requests that are supposed to use GET method, but in some particular case can't do it, because they have too long parameters. Although there may no longer be a limit on the length of the query string (I don't keep track of this).

Anyway my previous comment is about general Safe/Unsafe abstraction. But if you still want to give it Web-only meaning, then you do not need it at all. Abstraction layers are usually added to simplify the logic, not the other way around.

As I mentioned earlier, I will no longer try to push my vision of web API architecture, so you can just ignore my comments of this kind (unless I explicitly insist on something). They are more intended for other reviewers to show another vision for an alternative.